src/pages/workspace/tags/TagSettingsPage.tsx

```diff
@@ -0,0 +1,92 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useMemo} from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
+import OfflineWithFeedback from '@components/OfflineWithFeedback';
+import ScreenWrapper from '@components/ScreenWrapper';
+import Switch from '@components/Switch';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {setWorkspaceTagEnabled} from '@libs/actions/Policy';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import * as PolicyUtils from '@libs/PolicyUtils';
+import type {SettingsNavigatorParamList} from '@navigation/types';
+import NotFoundPage from '@pages/ErrorPage/NotFoundPage';
+import AdminPolicyAccessOrNotFoundWrapper from '@pages/workspace/AdminPolicyAccessOrNotFoundWrapper';
+import PaidPolicyAccessOrNotFoundWrapper from '@pages/workspace/PaidPolicyAccessOrNotFoundWrapper';
+import * as Policy from '@userActions/Policy';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type {PolicyTagList} from '@src/types/onyx';
+
+type TagSettingsPageOnyxProps = {
+    /** All policy tags */
+    policyTags: OnyxEntry<PolicyTagList>;
+};
+
+type TagSettingsPageProps = TagSettingsPageOnyxProps & StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.TAG_SETTINGS>;
+
+function TagSettingsPage({route, policyTags}: TagSettingsPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const policyTag = useMemo(() => PolicyUtils.getTagList(policyTags, 0), [policyTags]);
```

Comment:
Crash App from this line 


https://github.com/Expensify/App/assets/141406735/2afba092-010d-4808-948f-bb4e27911a5e